src/codegen.cpp

```diff
@@ -2796,7 +2864,11 @@ static void emit_assignment(jl_value_t *l, jl_value_t *r, jl_codectx_t *ctx)
                 rval = emit_unbox(vt->getContainedType(0), emit_unboxed(r, ctx), vi.declType);
             }
             else {
-                rval = boxed(emit_expr(r, ctx, true),ctx,rt);
+                rval = boxed(emit_expr(r, ctx, true), ctx, rt);
+                if (!is_stack(bp)) {
+                    Value* box = builder.CreateGEP(bp, ConstantInt::get(T_size, -1));
```

Comment:
oops, no, you are right. i forgot there is a branch in `emit_assignment` on whether this is a `jl_binding_t` or a `Box` 

> /me goes looking elsewhere for the source of his bug